### PR TITLE
feat: add cpp dictionary files

### PR DESCRIPTION
**Pull request by david-fong**
for #54 

I've left out my boost-library file for now since it's not finished yet.

Should I make any changes to the json files in this PR? Or can I leave that to you?

### CODE REVIEW

**Comment by Jason3S** [comment]
@david-fong,

Great, I'll pull these in.

Please note, it is not necessary to comment out short words like

```
# https://en.cppreference.com/w/cpp/header/compare
#eq
#neq
#lt
lteq
#gt
gteq
```

I think it is better to leave them in. It will not hurt performance.

File: dictionaries/cpp/src/stdlib-cpp.txt [hunk 1]
```diff
@@ -0,0 +1,509 @@
+# general
+errc
+
+
+# https://en.cppreference.com/w/cpp/header/version
+
+
+# https://en.cppreference.com/w/cpp/header/limits
+denorm
+signaling # signle "l" not a typo
```
Comment (Jason3S) [inline dictionaries/cpp/src/stdlib-cpp.txt:10]: but `signle` is a typo 😊

Comment (david-fong) [reply dictionaries/cpp/src/stdlib-cpp.txt:10]: hahaha nice catch xD.

Comment (david-fong) [reply dictionaries/cpp/src/stdlib-cpp.txt]: fixed

**Comment by Jason3S** [comment]
@david-fong,

I'll combine these into a new dictionary. Any suggestions on a name? `cpp-refined` was my first idea.

Note: Some people have the minimum word length set to 2, which is why it is useful to have the short words in the dictionary.
Note: Duplicate words get removed in the dictionary building process.

**Comment by Jason3S** [comment]
@david-fong,
Thank you.

**Comment by david-fong** [comment]
cpp-refined sounds good to me. cpp-dict-v2 could be another option. up to you.

my pleasure! I'm working on a c++ project so I'm benefitting from it too :)

I'll probably want to make some more future additions. Ex. some boost library words, or words for more compiler builtins, or very-popular libraries. Let me know if there's a procedure / format for how best to do that.

**Comment by Jason3S** [comment]
@david-fong,

I have wired everything up in #1102.

**Comment by david-fong** [comment]
wonderful! looking forward to the next release :)

**Comment by david-fong** [comment]
not sure where to say this. I just found out cppreference has a ["symbol index"](https://en.cppreference.com/w/cpp/symbol_index) page that looks like it could be useful as an information source for some kind of automation (as opposed to how I did things manually).

**Comment by Jason3S** [comment]
@david-fong,

That looks like a good resource. It might be possible with a `curl` command and a bit of parsing to generate a good dictionary.

This is their API: https://en.cppreference.com/mwiki/api.php
Here is the [symbol index JSON src](https://en.cppreference.com/mwiki/api.php?action=query&format=json&prop=revisions&titles=cpp/symbol_index&formatversion=2&rvprop=content&rvslots=*)

Since they are using the Creative Commons License, it is possible to use it.

**Comment by Jason3S** [comment]
> not sure where to say this. I just found out cppreference has a ["symbol index"](https://en.cppreference.com/w/cpp/symbol_index) page that looks like it could be useful as an information source for some kind of automation (as opposed to how I did things manually).

Just open a new issue to track using cppreference as a source.

**Pull request merged**